src/yb/util/wait_state.cc

```diff
@@ -22,6 +22,27 @@ namespace yb::util {
 // __thread WaitStateInfoPtr WaitStateInfo::threadlocal_wait_state_;
 thread_local WaitStateInfoPtr WaitStateInfo::threadlocal_wait_state_;
 
+void AUHMetadata::SetQueryIdFromHex(std::string query_id_str) {
+  int size = std::min(16, (int)query_id_str.length());
```

Comment:
Do we need to re-invent this?

Why not just use
stoi/stoul/boost::lexical_cast()

https://www.geeksforgeeks.org/different-ways-to-convert-hex-string-to-integer-in-cpp-stl/